### PR TITLE
use NetworkAccessManager in IceServer for single QNAM instance

### PR DESCRIPTION
**Pull request by birarda**
- fixes an issue with IceServer open file handles due to many QNetworkAccessManager instances

### CODE REVIEW

**Comment by birarda** [comment]
This is a fix for https://app.asana.com/0/11984398816645/101224810683313. Many thanks to @sethalves for sending me down the right path here.

I was incorrectly creating a QNetworkAccessManager per request for a domain's public key. Over time in the ice-server this would leave many open file handles and eventually the ice-server would lock up because it was no longer able to use epoll to process events.

Test this on linux. 
With the master ice-server running, watch `lsof` and notice that entries for the ice-server for `anon_inode` grow as you connect new domain-servers (you can re-create this by disconnecting, waiting for the ice-server to drop you, and reconnecting). On this PR you should see a few entries for `anon_inode` but they should not increase steadily as you connect/disconnect a domain-server.

You will need to re-compile a domain-server to change `ICE_SERVER_DEFAULT_HOSTNAME` on 45 of DomainServer.cpp to point to your test ice-server.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4210bc178e1302275a75fd2a7c8f6814e01facb3 available at: https://gist.github.com/bd5e05921fae9e6f143a6209fe410ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7513.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7513.dmg

**Comment by OmegaHeron** [comment]
@birarda - I'm compiling now and setting up to test this.  Will report back later, but, it'll be a bit of delay - getting all compiled and custom setup done.

**Comment by birarda** [comment]
@OmegaHeron I believe that @sethalves may be setting up to test as we speak

**Comment by OmegaHeron** [comment]
Good enough - won't burn time needlessly then.  Thanks.

**Comment by sethalves** [comment]
I've tried this on Linux -- the ice server functions and doesn't collect filedes as I reconnect to the domain.  It looks good to me.

File: ice-server/src/IceServer.cpp [hunk 1]
```diff
@@ -33,7 +34,8 @@ IceServer::IceServer(int argc, char* argv[]) :
     _id(QUuid::createUuid()),
     _serverSocket(),
     _activePeers(),
-    _httpManager(QHostAddress::AnyIPv4, ICE_SERVER_MONITORING_PORT, QString("%1/web/").arg(QCoreApplication::applicationDirPath()), this)
+    _httpManager(QHostAddress::AnyIPv4, ICE_SERVER_MONITORING_PORT, QString("%1/web/").arg(QCoreApplication::applicationDirPath()), this),
+	_lastInactiveCheckTimestamp(QDateTime::currentMSecsSinceEpoch())
```
Comment (Atlante45) [inline ice-server/src/IceServer.cpp]: Tab

File: ice-server/src/IceServer.cpp [hunk 2]
```diff
@@ -201,8 +201,8 @@ bool IceServer::isVerifiedHeartbeat(const QUuid& domainID, const QByteArray& pla
 
 void IceServer::requestDomainPublicKey(const QUuid& domainID) {
     // send a request to the metaverse API for the public key for this domain
-    QNetworkAccessManager* manager = new QNetworkAccessManager { this };
-    connect(manager, &QNetworkAccessManager::finished, this, &IceServer::publicKeyReplyFinished);
+    auto& networkAccessManager = NetworkAccessManager::getInstance();
```
Comment (Atlante45) [inline ice-server/src/IceServer.cpp]: That's why you don't use parenting for object cleanup ;)

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8b0dbf3bf8058c7ae0f85f18638fc64eb47c593 available at: https://gist.github.com/bd5e05921fae9e6f143a6209fe410ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7513.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7513.dmg

**Pull request merged**